saga/saga_state_test.go

```diff
@@ -55,26 +55,3 @@ func TestSagaState_Copy(t *testing.T) {
 		t.Error(fmt.Sprintf("Copy Should Preserve SagaId"))
 	}
 }
-
-func TestSagaState_SagaStateNotMutatedDuringUpdate(t *testing.T) {
```

Comment:
this should be replaced with a test that ensures that an update resulting in an error does not result in a mutation